src/docs/release/breaking-changes/12_18_19_ImageCache_Logic.md

```diff
@@ -0,0 +1,69 @@
+---
+title: Revert ImageCache algorithm for large image
+description: Stop increasing the ImageCache maxByteSize to accomodate large
+images.
+---
+
+# Revert ImageCache algorithm for large image
+
+## Context
+
+When loading images into the ImageCache whose byte size is larger than the
+ImageCache's maxByteSize we previously would permanently increase the
+maxByteSize to accommodate them.  This logic can lead to creeping maxByteSize
+which makes working in memory limited systems more difficult.
+
+## Description of change
+
+The following changes were made to the ImageCache algorithm:
+
+```dart
+// Old Logic Pseudocode
+void onLoadImage(Image image) {
+  if (image.byteSize > _cache.maxByteSize) {
+    _cache.maxByteSize = image.byteSize + 1000;
+  }
+  _cache.add(image);
+  while (_cache.count > _cache.maxCount 
+      || _cache.byteSize > _cache.maxByteSize) {
+    _cache.discardOldestImage();
+  }
+}
+```
+
+```dart
+// New Logic Pseudocode
+void onLoadImage(Image image) {
+  if (image.byteSize < _cache.maxByteSize) {
+    _cache.add(image);
+    while (_cache.count > _cache.maxCount 
+        || _cache.byteSize > cache.maxByteSize) {
+      cache.discardOldestImage();
+    }
+  }
+}
+```
+
+## Migration guide
+
+There may be situations where the ImageCache is thrashing with the new logic
+where it wasn't previously, specifically if you are loading images that are
+larger than your cache.maxByteSize.  These can be remedied by one of the
```

Comment:
Does this cause thrash, or just mean you won't cache an image that you used to cache?

Reply:
I think it is unlikely that someone will get thrash where they didn't previously.  You would have to be causing build to happen regularly and also have one large image and small images that fit in 1000 bytes methinks.

Reply:
But thrashing would be the case where they'd want to address the change.